src/components/performances/PerformancesSection.jsx

```diff
@@ -44,13 +44,22 @@ class PerformancesSection extends React.Component {
       return res;
     }, []);
 
-  // TODO: implement filters for award considerations and search
-  handleFilters = (filters) => {
+  // TODO: implement filters for award considerations
+  handleFilters = (filters, search) => {
     const { performances } = this.state;
     const keys = this.getFilterKeys(filters);
-    const filteredPerformances = performances.filter(
-      performance => keys.every(key => filters[key].includes(performance[key]))
-    );
+    const filterFunction = (performance) => {
+      const danceTitle = performance.danceTitle.toLowerCase();
+      const passFilter = keys.every(key => filters[key].includes(performance[key]));
+
+      if (search && search.length > 0) {
```

Comment:
Could we rename some of these variables to be more descriptive. I'm not too sure what "passFilter" is and "search" is kind of generic.

Also, it would be really nice to have some comments especially here. This block of code is a bit less intuitive.